user-interface/src/main/java/life/qbic/datamanager/views/projects/project/measurements/registration/MeasurementUpload.java

```diff
@@ -0,0 +1,424 @@
+import life.qbic.datamanager.views.general.upload.EditableMultiFileMemoryBuffer;
+import life.qbic.datamanager.views.notifications.MessageSourceNotificationFactory;
+import life.qbic.projectmanagement.application.ValidationResult;
+import life.qbic.projectmanagement.application.api.AsyncProjectService;
+import life.qbic.projectmanagement.application.api.AsyncProjectService.ValidationRequest;
+import life.qbic.projectmanagement.application.api.AsyncProjectService.ValidationRequestBody;
+import life.qbic.projectmanagement.application.api.AsyncProjectService.ValidationResponse;
+import reactor.core.publisher.Flux;
+
+/**
+ * <b>MeasurementUpload</b>
+ * <p>
+ * A data manager component that enables users to upload measurement files to register or update
+ * measurement metadata.
+ * <p>
+ * The component is created with a certain type of {@link MetadataConverterV2}, such that the
+ * content gets parsed and validated immediately after the upload.
+ *
+ * @since 1.11.0
+ */
+public class MeasurementUpload extends Div implements UserInput {
+
+  public static final int MAX_FILE_SIZE_BYTES = (int) (Math.pow(1024, 2) * 16);
+  private final UploadItemsDisplay uploadItemsDisplay;
+  private final AsyncProjectService service;
+  private final Context context;
+  private final Div validationProgress;
+  private final MetadataConverterV2<? extends ValidationRequestBody> converter;
+
+  private final List<? extends ValidationRequestBody> validationRequests = new ArrayList<>();
+  private final MessageSourceNotificationFactory notificationFactory;
+
+  private enum AcceptedFileType {
+    EXCEL("application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"),
+    TSV("text/tab-separated-values", "text/plain");
+    private final List<String> mimeTypes;
+
+    AcceptedFileType(String... mimeTypes) {
+      this.mimeTypes = List.of(mimeTypes);
+    }
+
+    static Optional<AcceptedFileType> forMimeType(String mimeType) {
+      return Arrays.stream(values())
+          .filter(it -> it.mimeTypes.contains(mimeType))
+          .findFirst();
+    }
+
+    static Set<String> allMimeTypes() {
+      return Arrays.stream(values()).flatMap(it -> it.mimeTypes.stream())
+          .collect(Collectors.toUnmodifiableSet());
+    }
+  }
+
+  private final EditableMultiFileMemoryBuffer uploadBuffer;
+  private final List<MeasurementFileItem> measurementFileItems;
+
+  public MeasurementUpload(
+      AsyncProjectService service,
+      Context context,
+      MetadataConverterV2<? extends ValidationRequestBody> converter,
+      MessageSourceNotificationFactory notificationFactory) {
+    this.uploadBuffer = new EditableMultiFileMemoryBuffer();
+    this.measurementFileItems = new ArrayList<>();
+    this.service = requireNonNull(service);
+    this.context = requireNonNull(context);
+    this.validationProgress = new Div();
+    this.converter = requireNonNull(converter);
+    this.notificationFactory = requireNonNull(notificationFactory);
```

Comment:
NIT: I find this section hard to read. Mainly because it is unclear where the method parameters are assigned. I suggest reordering the statements to first have the assignments like `this.service = requireNonNull(service)` and then having the internal fields initialized e.g. `this.validationProgress = new Div()`
This might be a personal preference so feel free to ignore it if you do not agree.